pkg/common/health/cache_test.go

```diff
@@ -80,25 +79,98 @@ func TestHealthFailsAndRecover(t *testing.T) {
 	err = c.addCheck("bar", barChecker)
 	require.NoError(t, err)
 
-	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
-	defer cancel()
+	ctx := context.Background()
 
 	err = c.start(ctx)
 	require.NoError(t, err)
 
-	t.Run("start successfully", func(t *testing.T) {
+	t.Run("fail to start initially", func(t *testing.T) {
 		// Wait for initial calls
-		select {
-		case <-waitFor:
-		case <-ctx.Done():
-			require.Fail(t, "unable to get updates because context is finished")
-		}
+		<-waitFor
```

Comment:
is no possible that we get into a blocked test here?
or the plan is to depends in test timeout instead?

Reply:
In the past we decided to always rely on `go test` timeouts rather than introducing test-specific timeouts, which might be unreliable. https://github.com/spiffe/spire/pull/2121